bundles/org.openhab.core.config.discovery.addon/src/main/java/org/openhab/core/config/discovery/addon/AddonSuggestionService.java

```diff
@@ -161,43 +161,61 @@ private boolean isFinderEnabled(AddonFinder finder) {
 
     @Reference(cardinality = ReferenceCardinality.MULTIPLE, policy = ReferencePolicy.DYNAMIC)
     public void addAddonInfoProvider(AddonInfoProvider addonInfoProvider) {
-        addonInfoProviders.add(addonInfoProvider);
+        synchronized (this) {
+            addonInfoProviders.add(addonInfoProvider);
+        }
         changed();
     }
 
     public void removeAddonInfoProvider(AddonInfoProvider addonInfoProvider) {
-        if (addonInfoProviders.remove(addonInfoProvider)) {
+        boolean removed;
+        synchronized (this) {
+            removed = addonInfoProviders.remove(addonInfoProvider);
+        }
+        if (removed) {
             changed();
         }
     }
 
     @Reference(cardinality = ReferenceCardinality.MULTIPLE, policy = ReferencePolicy.DYNAMIC)
     public void addAddonFinder(AddonFinder addonFinder) {
-        addonFinders.add(addonFinder);
+        synchronized (this) {
+            addonFinders.add(addonFinder);
+        }
         changed();
     }
 
     public void removeAddonFinder(AddonFinder addonFinder) {
-        if (addonFinders.remove(addonFinder)) {
+        boolean removed;
+        synchronized (this) {
+            removed = addonFinders.remove(addonFinder);
+        }
+        if (removed) {
             changed();
         }
     }
 
     private void changed() {
-        List<AddonInfo> candidates = addonInfoProviders.stream().map(p -> p.getAddonInfos(localeProvider.getLocale()))
-                .flatMap(Collection::stream).toList();
-        addonFinders.stream().filter(this::isFinderEnabled).forEach(f -> f.setAddonCandidates(candidates));
+        synchronized (this) {
+            List<AddonInfo> candidates = addonInfoProviders.stream()
+                    .map(p -> p.getAddonInfos(localeProvider.getLocale())).flatMap(Collection::stream).toList();
+            addonFinders.stream().filter(this::isFinderEnabled).forEach(f -> f.setAddonCandidates(candidates));
+        }
     }
 
     @Deactivate
     @Override
     public void close() throws Exception {
-        addonFinders.clear();
-        addonInfoProviders.clear();
+        synchronized (this) {
+            addonFinders.clear();
+            addonInfoProviders.clear();
+        }
     }
 
     public Set<AddonInfo> getSuggestedAddons(@Nullable Locale locale) {
-        return addonFinders.stream().filter(this::isFinderEnabled).map(f -> f.getSuggestedAddons())
-                .flatMap(Collection::stream).collect(Collectors.toSet());
+        synchronized (this) {
```

Comment:
Perhaps you can simply synchronize on the object itself?
```suggestion
        synchronized (addonFinders) {
```

Reply:
Ok.